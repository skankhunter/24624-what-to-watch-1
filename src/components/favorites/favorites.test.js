import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";


import Favorites from "./favorites.jsx";

describe(`Favorites:`, () => {
  it(`Correctly renders after relaunch`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <Favorites
              authorized={true}
              userAvatar={`ava`}
              userName={`Dick`}
            />
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
