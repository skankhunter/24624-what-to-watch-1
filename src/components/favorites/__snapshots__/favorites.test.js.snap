// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites: Correctly renders after relaunch 1`] = `
Array [
  <div
    className="visually-hidden"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        id="add"
        viewBox="0 0 19 20"
      >
        <title>
          +
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Page-1"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            id="+"
            points="10.777832 11.2880859 10.777832 19.5527344 8.41650391 19.5527344 8.41650391 11.2880859 0.627929688 11.2880859 0.627929688 8.92675781 8.41650391 8.92675781 8.41650391 0.662109375 10.777832 0.662109375 10.777832 8.92675781 18.5664062 8.92675781 18.5664062 11.2880859"
          />
        </g>
      </symbol>
      <symbol
        id="full-screen"
        viewBox="0 0 27 27"
      >
        <path
          clipRule="evenodd"
          d="M23.8571 0H16V3.14286H23.8571V11H27V3.14286V0H23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M27 23.8571V16H23.8571V23.8571H16V27H23.8571H27L27 23.8571Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M0 3.14286L0 11H3.14286L3.14286 3.14286L11 3.14286V0H3.14286H0L0 3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M3.14286 27H11V23.8571H3.14286L3.14286 16H0L0 23.8571V27H3.14286Z"
          fill="#FFF9D9"
          fillOpacity="0.7"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="in-list"
        viewBox="0 0 18 14"
      >
        <path
          clipRule="evenodd"
          d="M2.40513 5.35353L6.1818 8.90902L15.5807 0L18 2.80485L6.18935 14L0 8.17346L2.40513 5.35353Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
      <symbol
        id="pause"
        viewBox="0 0 14 21"
      >
        <title>
          Artboard
        </title>
        <desc>
          Created with Sketch.
        </desc>
        <g
          fill="none"
          fillRule="evenodd"
          id="Artboard"
          stroke="none"
          strokeWidth="1"
        >
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="0 -1.11910481e-13 4 -1.11910481e-13 4 21 0 21"
          />
          <polygon
            fill="#EEE5B5"
            fillRule="nonzero"
            id="Line"
            points="10 -1.11910481e-13 14 -1.11910481e-13 14 21 10 21"
          />
        </g>
      </symbol>
      <symbol
        id="play-s"
        viewBox="0 0 19 19"
      >
        <path
          clipRule="evenodd"
          d="M0 0L19 9.5L0 19V0Z"
          fill="#EEE5B5"
          fillRule="evenodd"
        />
      </symbol>
    </svg>
  </div>,
  <div
    className="user-page"
  >
    <header
      className="page-header user-page__head"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        className="page-title user-page__title"
      >
        My list
      </h1>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <section
      className="catalog"
    >
      <h2
        className="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <div
        className="catalog__movies-list"
      >
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Fantastic Beasts: The Crimes of Grindelwald"
              height="175"
              src="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Fantastic Beasts: The Crimes of Grindelwald
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Bohemian Rhapsody"
              height="175"
              src="img/bohemian-rhapsody.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Bohemian Rhapsody
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Macbeth"
              height="175"
              src="img/macbeth.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Macbeth
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Aviator"
              height="175"
              src="img/aviator.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Aviator
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="We need to talk about Kevin"
              height="175"
              src="img/we-need-to-talk-about-kevin.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              We need to talk about Kevin
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="What We Do in the Shadows"
              height="175"
              src="img/what-we-do-in-the-shadows.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              What We Do in the Shadows
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Revenant"
              height="175"
              src="img/revenant.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Revenant
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Johnny English"
              height="175"
              src="img/johnny-english.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Johnny English
            </a>
          </h3>
        </article>
        <article
          className="small-movie-card catalog__movies-card"
        >
          <button
            className="small-movie-card__play-btn"
            type="button"
          >
            Play
          </button>
          <div
            className="small-movie-card__image"
          >
            <img
              alt="Shutter Island"
              height="175"
              src="img/shutter-island.jpg"
              width="280"
            />
          </div>
          <h3
            className="small-movie-card__title"
          >
            <a
              className="small-movie-card__link"
              href="movie-page.html"
            >
              Shutter Island
            </a>
          </h3>
        </article>
      </div>
    </section>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="main.html"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>,
]
`;
